sdks/java/extensions/zetasketch/src/test/java/org/apache/beam/sdk/extensions/zetasketch/HllCountTest.java

```diff
@@ -0,0 +1,373 @@
+
+    PAssert.that(result).containsInAnyOrder(Collections.singletonList(KV.of("k", STRINGS_SKETCH)));
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testInitForStringsPerKey_WithPrecision() {
+    List<KV<String, String>> input = new ArrayList<>();
+    STRINGS.forEach(s -> input.add(KV.of("k", s)));
+    PCollection<KV<String, byte[]>> result =
+        p.apply(Create.of(input))
+            .apply(HllCount.Init.forStrings().withPrecision(TEST_PRECISION).perKey());
+
+    PAssert.that(result)
+        .containsInAnyOrder(Collections.singletonList(KV.of("k", STRINGS_SKETCH_TEST_PRECISION)));
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testInitForStringsPerKey_WithInvalidPrecision() {
+    List<KV<String, String>> input = new ArrayList<>();
+    STRINGS.forEach(s -> input.add(KV.of("k", s)));
+    thrown.expect(IllegalArgumentException.class);
+    p.apply(Create.of(input)).apply(HllCount.Init.forStrings().withPrecision(0).perKey());
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testInitForBytesPerKey() {
+    List<KV<String, byte[]>> input = new ArrayList<>();
+    BYTES.forEach(bs -> input.add(KV.of("k", bs)));
+    PCollection<KV<String, byte[]>> result =
+        p.apply(Create.of(input)).apply(HllCount.Init.forBytes().perKey());
+
+    PAssert.that(result).containsInAnyOrder(Collections.singletonList(KV.of("k", BYTES_SKETCH)));
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testMergePartialGlobally() {
+    PCollection<byte[]> result =
+        p.apply(Create.of(INTS1_SKETCH, INTS2_SKETCH)).apply(HllCount.MergePartial.globally());
+
+    PAssert.thatSingleton(result).isEqualTo(INTS1_INTS2_SKETCH);
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testMergePartialGlobally_MergeWithSketchForEmptySet() {
+    PCollection<byte[]> result =
+        p.apply(Create.of(LONGS_SKETCH, LONGS_EMPTY_SKETCH))
+            .apply(HllCount.MergePartial.globally());
+
+    PAssert.thatSingleton(result).isEqualTo(LONGS_SKETCH);
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testMergePartialGlobally_SingletonInput() {
+    PCollection<byte[]> result =
+        p.apply(Create.of(LONGS_SKETCH)).apply(HllCount.MergePartial.globally());
+
+    PAssert.thatSingleton(result).isEqualTo(LONGS_SKETCH);
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testMergePartialGlobally_EmptyInput() {
+    PCollection<byte[]> result =
+        p.apply(Create.empty(TypeDescriptor.of(byte[].class)))
+            .apply(HllCount.MergePartial.globally());
+
+    PAssert.that(result).empty();
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testMergePartialGlobally_IncompatibleSketches() {
+    p.apply(Create.of(INTS1_SKETCH, STRINGS_SKETCH)).apply(HllCount.MergePartial.globally());
+
+    thrown.expect(PipelineExecutionException.class);
+    p.run();
+  }
```

Comment:
Another corner case you could add is to merge STRINGS_SKETCH and STRINGS_SKETCH_TEST_PRECISION, but I do expect that to just go through, so ... :)

Reply:
Yeah, but I think this will then become a wrapper around the `merge_NormalIntoNormalWith[Higher|Lower]Precision` [tests](https://github.com/google/zetasketch/blob/4aa44e9cb543d766318b919fc6bb359bdf7b0809/javatests/com/google/zetasketch/HyperLogLogPlusPlusTest.java#L733), and the expected result is just that it goes through, as you mentioned. So I would like to skip it here. Let me know if you disagree.